hedera-node/hedera-mono-service/src/main/java/com/hedera/services/contracts/execution/EvmTxProcessor.java

```diff
@@ -336,4 +337,48 @@ private ChargingResult chargeForGas(
         }
         return new ChargingResult(mutableSender, mutableRelayer, allowanceCharged);
     }
+
+    private void handleResourceLimitExceeded(
+            final Account sender,
+            final long gasPrice,
+            final long gasLimit,
+            final long value,
+            final boolean isStatic,
+            final BigInteger userOfferedGasPrice,
+            final long maxGasAllowanceInTinybars,
+            final Account relayer,
+            final Wei gasCost,
+            final Wei upfrontCost) {
+        // Consume all gas on resource exhaustion, using a clean updater
+        final var feesOnlyUpdater = (HederaWorldState.Updater) worldState.updater();
+        chargeForGas(
+                gasCost,
+                upfrontCost,
+                value,
+                maxGasAllowanceInTinybars,
+                intrinsicGas,
+                gasPrice,
+                gasLimit,
+                isStatic,
+                userOfferedGasPrice,
+                sender.getId().asEvmAddress(),
+                relayer == null ? null : relayer.getId().asEvmAddress(),
+                feesOnlyUpdater);
+        sendToCoinbase(coinbase, gasLimit, gasPrice, feesOnlyUpdater);
+        // We can't go through the top-level commit() because that would
+        // re-try to commit the storage changes
+        feesOnlyUpdater.trackingAccounts().commit();
+    }
+
+    private TransactionProcessingResult createResourceLimitExceededResult(
+            final long gasPrice, final long gasLimit, final ResourceLimitException e) {
+        return TransactionProcessingResult.failed(
```

Comment:
(it's a shame `TransactionProcessingResult` doesn't have a builder ... all these empty parameters and the reader either has to _know_ what they are or go look it up)